auslib/admin/views/forms.py

```diff
@@ -265,3 +295,41 @@ class ReadOnlyForm(Form):
     product = StringField('Product', validators=[InputRequired()])
     read_only = BooleanField('read_only')
     data_version = IntegerField('data_version', widget=HiddenInput())
+
+
+class ScheduledChangeNewReleaseForm(ScheduledChangeTimeForm):
+    name = StringField('Name', validators=[InputRequired()])
+    product = StringField('Product', validators=[InputRequired()])
+    data = JSONStringField({}, 'Data', validators=[InputRequired()], widget=FileInput())
+    data_version = IntegerField('data_version', widget=HiddenInput())
+    change_type = SelectField("Change Type", choices=[('insert', 'insert'), ('update', 'update'), ('delete'), ('delete')])
+
+
+class ScheduledChangeExistingReleaseForm(ScheduledChangeTimeForm):
+    name = StringField('Name', validators=[InputRequired()])
+    product = StringField('Product', validators=[Optional()])
+    data = JSONStringField({}, 'Data', validators=[Optional()], widget=FileInput())
+    data_version = IntegerField('data_version', widget=HiddenInput())
```

Comment:
Great questions. This is something we didn't need worry about for Rule Scheduled Changes, I suppose, because the data is simpler and the PK is hidden and unchangeable.

The reason name is required here isn't to allow for changing it - it's required because we need to know which Release to update. Unlike when editing a Release directly, it's not included in the URL for Scheduled Changes, so we need to put it in the Form. I agree that changing the Product of an existing Release is unlikely (and we can always do that by deleting/adding a new one, if we need to). How about this for Releases:
* For Scheduled Changes that create new Releases:
** When creating, name, product, and data will be required, matching the db rules. Neither data_version nor sc_data_version will be allowed.
** When editing, name, product, and data will be optional. We update based on sc_id, so we don't need to know name to edit, but we should allow for other base table fields to be passed in case there's a typo fix or other need to change the name. Same thing for product & data. sc_data_version will be required, data_version will not be allowed.
* For Scheduled Changes that change existing Releases:
** When creating, name and data will be required. Product will not be allowed, per above about it being confusing to allow product to change. data_version will be required, sc_data_version will not be allowed.
** When editing, only data will be allowed. sc_data_version will be required, data_version will not be allowed.

I think Permissions may be similar, too, except we'd treat both permission and username like we treat name, and treat options like data. Spelled out more, this means:
* For Scheduled Changes that create new Permissions:
** When creating, permission, username, and options will be required. Neither data_version nor sc_data_version will be allowed.
** When editing, permission, username, and options will all be optionaltd. sc_data_version will be required, data_version will not be allowed.
* For Scheduled Changes that modify a Permission:
** When creating, permission, username, and options will be required. data_version will be required, sc_data_version will not be allowed.
** When editing, only options will be allowed. sc_data_version will be required, data_version will not be allowed.